### PR TITLE
conjure-java codegen uses the palantir java formatter

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
conjure-java codegen uses the palantir java formatter
==COMMIT_MSG==

### CODE REVIEW

File: conjure-java-core/src/integrationInput/java/com/palantir/product/AsyncRequestProcessingTestServiceEndpoints.java [hunk 1]
```diff
@@ -38,17 +37,15 @@ public static UndertowService of(UndertowAsyncRequestProcessingTestService deleg
...
-                        new ThrowsInHandlerEndpoint(runtime, delegate),
-                        new FailedFutureEndpoint(runtime, delegate),
-                        new BinaryEndpoint(runtime, delegate),
-                        new FutureTraceIdEndpoint(runtime, delegate)));
+        return Collections.unmodifiableList(Arrays.asList(
+                new DelayEndpoint(runtime, delegate),
+                new ThrowsInHandlerEndpoint(runtime, delegate),
+                new FailedFutureEndpoint(runtime, delegate),
+                new BinaryEndpoint(runtime, delegate),
+                new FutureTraceIdEndpoint(runtime, delegate)));
```
Comment (carterkozak) [inline conjure-java-core/src/integrationInput/java/com/palantir/product/AsyncRequestProcessingTestServiceEndpoints.java:45]: this is a lot cleaner

File: conjure-java-core/src/integrationInput/java/com/palantir/product/EteService.java [hunk 2]
```diff
@@ -29,21 +29,20 @@
...
      * <h2>Very Important Documentation</h2>
-     *
-     * <p>This documentation provides a <em>list</em>:
-     *
+     * <p>This documentation provides a <em>list</em>:</p>
      * <ul>
-     *   <li>Docs rule
-     *   <li>Lists are wonderful
+     * <li>Docs rule</li>
+     * <li>Lists are wonderful</li>
```
Comment (carterkozak) [inline conjure-java-core/src/integrationInput/java/com/palantir/product/EteService.java:36]: huh, at one point I think we had an option to eagerly format javadoc. @dansanduleac do we still have a flag for that?

**Comment by ferozco** [comment]
@carterkozak having a bit of trouble spotting the actual code change

File: conjure-java-core/build.gradle [hunk 3]
```diff
@@ -43,7 +43,7 @@ dependencies {
     compile 'com.fasterxml.jackson.core:jackson-databind'
     compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
     compile 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8'
-    compile 'com.google.googlejavaformat:google-java-format'
+    compile 'com.palantir.javaformat:palantir-java-format'
```
Comment (carterkozak) [inline conjure-java-core/build.gradle:46]: code change

File: conjure-java-core/src/main/java/com/palantir/conjure/java/util/Goethe.java [hunk 4]
```diff
@@ -37,8 +37,8 @@
 /** Tools for a better JavaPoet. */
 public final class Goethe {
 
-    private static final Formatter JAVA_FORMATTER = new Formatter(JavaFormatterOptions.builder()
-            .style(JavaFormatterOptions.Style.AOSP)
+    private static final Formatter JAVA_FORMATTER = Formatter.createFormatter(JavaFormatterOptions.builder()
+            .style(JavaFormatterOptions.Style.PALANTIR)
```
Comment (carterkozak) [inline conjure-java-core/src/main/java/com/palantir/conjure/java/util/Goethe.java:41]: code change

**Comment by carterkozak** [comment]
@ferozco I've added comments on the two places with human changes.

File: conjure-java-core/build.gradle (hunk 3 above)
Comment (ferozco) [reply conjure-java-core/build.gradle]: I didn't realize that we service loaded the formatter. That's cool!

**Comment by ferozco** [comment]
👍

**Pull request merged**